src/test/java/ch/uzh/ifi/hase/soprafs23/service/UserServiceTest.java

```diff
@@ -21,65 +23,74 @@ public class UserServiceTest {
+    User createdUser = userService.createUser(user);
 
     // then
-    Mockito.verify(userRepository, Mockito.times(1)).save(Mockito.any());
+    verify(userRepository, times(1)).save(Mockito.any());
+    verify(userRepository, times(1)).flush();
 
-    assertEquals(testUser.getId(), createdUser.getId());
-    assertEquals(testUser.getName(), createdUser.getName());
-    assertEquals(testUser.getUsername(), createdUser.getUsername());
+    assertEquals(createdUser.getId(), user.getId());
+    assertEquals(createdUser.getUsername(), user.getUsername());
+    assertEquals(createdUser.getPassword(), user.getPassword());
     assertNotNull(createdUser.getToken());
-    assertEquals(UserStatus.OFFLINE, createdUser.getStatus());
+    assertEquals(createdUser.getStatus(), UserStatus.ONLINE);
   }
 
   @Test
-  public void createUser_duplicateName_throwsException() {
-    // given -> a first user has already been created
-    userService.createUser(testUser);
+  public void createUser_duplicateUsername_throwsException() {
+    // a first user has already been created
+    userService.createUser(user);
 
-    // when -> setup additional mocks for UserRepository
-    Mockito.when(userRepository.findByName(Mockito.any())).thenReturn(testUser);
-    Mockito.when(userRepository.findByUsername(Mockito.any())).thenReturn(null);
+    // mocks findByUsername(username) method in UserRepository
+    Mockito.when(userRepository.findByUsername(user.getUsername())).thenReturn(user);
 
-    // then -> attempt to create second user with same user -> check that an error
-    // is thrown
-    assertThrows(ResponseStatusException.class, () -> userService.createUser(testUser));
+    // then
+    assertThrows(ResponseStatusException.class, () -> userService.createUser(user));
   }
 
   @Test
-  public void createUser_duplicateInputs_throwsException() {
-    // given -> a first user has already been created
-    userService.createUser(testUser);
+  public void createUser_invalidUsername_throwsException() {
+    user.setUsername("invalid-username");
 
-    // when -> setup additional mocks for UserRepository
-    Mockito.when(userRepository.findByName(Mockito.any())).thenReturn(testUser);
-    Mockito.when(userRepository.findByUsername(Mockito.any())).thenReturn(testUser);
+    assertThrows(ResponseStatusException.class, () -> userService.createUser(user));
+  }
+
+  @Test
+  public void createUser_equalUsernameAndPassword_throwsException() {
+    user.setUsername("word");
+    user.setPassword("word");
 
-    // then -> attempt to create second user with same user -> check that an error
-    // is thrown
-    assertThrows(ResponseStatusException.class, () -> userService.createUser(testUser));
+    assertThrows(ResponseStatusException.class, () -> userService.createUser(user));
   }
 
+  @Test
+  public void loginUser_NameNotExists_throwsException() {
+
+      assertThrows(ResponseStatusException.class, () -> userService.getUserByUsername(null));
+    }
+
+    @Test
```

Comment:
why is this test indented?

Reply:
just forgot to press automatic formatting ;)